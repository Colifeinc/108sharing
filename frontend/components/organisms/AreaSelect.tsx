import * as React from 'react'
import I18n from 'core/i18n'
import { css } from '@emotion/core'

interface IProps {
  title: string
}

const AreaSelect: React.FC<IProps> = props => {
  return (
    <div>
      <h1 className="text-2xl font-bold mb-6">{props.title}</h1>

      <ul className="flex flex-wrap lg:flex-row -my-1 -mx-2">
        {['tokyo', 'osaka', 'kyoto', 'sapporo'].map(key => (
          <li className="w-1/2 lg:w-1/4 py-1 px-2" key={key}>
            <div className="relative w-full h-50 lg:h-90 flex items-center justify-center rounded-md">
              <a className="block" href={`/posts/search?area=${I18n.t(`search_pan.${key}`)}`}>
                <div className="absolute bottom-0 inset-x-0 flex items-end justify-center text-white z-1 text-2xl font-bold mb-6">
                  <span>{I18n.t(`search_pan.${key}`)}</span>
                </div>
                <img
                  className="absolute top-0 left-0 w-full h-full object-cover"
                  src={`/images/top/img_${key}.jpg`}
                  alt=""
                />
                <div
                  css={css`
                    position: absolute;
                    background-image: linear-gradient(
                      0deg,
                      #000000 0%,
                      rgba(0, 0, 0, 0.82) 52.08%,
                      rgba(0, 0, 0, 0) 100%
                    );
                    opacity: 0.52;
                    height: 25%;
                    left: 0;
                    right: 0;
                    bottom: 0;
                  `}
                />
              </a>
            </div>
          </li>
        ))}
      </ul>
    </div>
  )
}

export default AreaSelect
