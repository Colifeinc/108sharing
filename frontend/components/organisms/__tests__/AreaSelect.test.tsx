import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import AreaSelect from '../AreaSelect'

jest.mock('../../../core/i18n')

describe('organisms/AreaSelect', () => {
  it('should matches to the snapshot', () => {
    const wrapper = shallow(<AreaSelect title="title" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
