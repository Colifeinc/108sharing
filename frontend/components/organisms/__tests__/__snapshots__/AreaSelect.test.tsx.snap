// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/AreaSelect should matches to the snapshot 1`] = `
<div>
  <h1
    className="text-2xl font-bold mb-6"
  >
    title
  </h1>
  <ul
    className="flex flex-wrap lg:flex-row -my-1 -mx-2"
  >
    <li
      className="w-1/2 lg:w-1/4 py-1 px-2"
      key="tokyo"
    >
      <div
        className="relative w-full h-50 lg:h-90 flex items-center justify-center rounded-md"
      >
        <a
          className="block"
          href="/posts/search?area=Tokyo"
        >
          <div
            className="absolute bottom-0 inset-x-0 flex items-end justify-center text-white z-1 text-2xl font-bold mb-6"
          >
            <span>
              Tokyo
            </span>
          </div>
          <img
            alt=""
            className="absolute top-0 left-0 w-full h-full object-cover"
            src="/images/top/img_tokyo.jpg"
          />
          <div
            css={
              Object {
                "map": undefined,
                "name": "1vhbbox",
                "next": undefined,
                "styles": "
                    position: absolute;
                    background-image: linear-gradient(
                      0deg,
                      #000000 0%,
                      rgba(0, 0, 0, 0.82) 52.08%,
                      rgba(0, 0, 0, 0) 100%
                    );
                    opacity: 0.52;
                    height: 25%;
                    left: 0;
                    right: 0;
                    bottom: 0;
                  ",
                "toString": [Function],
              }
            }
          />
        </a>
      </div>
    </li>
    <li
      className="w-1/2 lg:w-1/4 py-1 px-2"
      key="osaka"
    >
      <div
        className="relative w-full h-50 lg:h-90 flex items-center justify-center rounded-md"
      >
        <a
          className="block"
          href="/posts/search?area=Osaka"
        >
          <div
            className="absolute bottom-0 inset-x-0 flex items-end justify-center text-white z-1 text-2xl font-bold mb-6"
          >
            <span>
              Osaka
            </span>
          </div>
          <img
            alt=""
            className="absolute top-0 left-0 w-full h-full object-cover"
            src="/images/top/img_osaka.jpg"
          />
          <div
            css={
              Object {
                "map": undefined,
                "name": "1vhbbox",
                "next": undefined,
                "styles": "
                    position: absolute;
                    background-image: linear-gradient(
                      0deg,
                      #000000 0%,
                      rgba(0, 0, 0, 0.82) 52.08%,
                      rgba(0, 0, 0, 0) 100%
                    );
                    opacity: 0.52;
                    height: 25%;
                    left: 0;
                    right: 0;
                    bottom: 0;
                  ",
                "toString": [Function],
              }
            }
          />
        </a>
      </div>
    </li>
    <li
      className="w-1/2 lg:w-1/4 py-1 px-2"
      key="kyoto"
    >
      <div
        className="relative w-full h-50 lg:h-90 flex items-center justify-center rounded-md"
      >
        <a
          className="block"
          href="/posts/search?area=Kyoto"
        >
          <div
            className="absolute bottom-0 inset-x-0 flex items-end justify-center text-white z-1 text-2xl font-bold mb-6"
          >
            <span>
              Kyoto
            </span>
          </div>
          <img
            alt=""
            className="absolute top-0 left-0 w-full h-full object-cover"
            src="/images/top/img_kyoto.jpg"
          />
          <div
            css={
              Object {
                "map": undefined,
                "name": "1vhbbox",
                "next": undefined,
                "styles": "
                    position: absolute;
                    background-image: linear-gradient(
                      0deg,
                      #000000 0%,
                      rgba(0, 0, 0, 0.82) 52.08%,
                      rgba(0, 0, 0, 0) 100%
                    );
                    opacity: 0.52;
                    height: 25%;
                    left: 0;
                    right: 0;
                    bottom: 0;
                  ",
                "toString": [Function],
              }
            }
          />
        </a>
      </div>
    </li>
    <li
      className="w-1/2 lg:w-1/4 py-1 px-2"
      key="sapporo"
    >
      <div
        className="relative w-full h-50 lg:h-90 flex items-center justify-center rounded-md"
      >
        <a
          className="block"
          href="/posts/search?area=Sapporo"
        >
          <div
            className="absolute bottom-0 inset-x-0 flex items-end justify-center text-white z-1 text-2xl font-bold mb-6"
          >
            <span>
              Sapporo
            </span>
          </div>
          <img
            alt=""
            className="absolute top-0 left-0 w-full h-full object-cover"
            src="/images/top/img_sapporo.jpg"
          />
          <div
            css={
              Object {
                "map": undefined,
                "name": "1vhbbox",
                "next": undefined,
                "styles": "
                    position: absolute;
                    background-image: linear-gradient(
                      0deg,
                      #000000 0%,
                      rgba(0, 0, 0, 0.82) 52.08%,
                      rgba(0, 0, 0, 0) 100%
                    );
                    opacity: 0.52;
                    height: 25%;
                    left: 0;
                    right: 0;
                    bottom: 0;
                  ",
                "toString": [Function],
              }
            }
          />
        </a>
      </div>
    </li>
  </ul>
</div>
`;
